### PR TITLE
projects:iio_demo: Makefile corrected

### PR DESCRIPTION
**Pull request by APorumb**
-corrected mistakes done in PR #1128

Signed-off-by: Andrei Porumb <andrei.porumb@analog.com>

### CODE REVIEW

**Review by mchindri — COMMENTED** [review]
Can you explain more about the mistake?

File: projects/iio_demo/src.mk [hunk 1]
```diff
@@ -10,8 +10,7 @@ SRCS +=	$(NO-OS)/util/list.c \
 SRCS += $(DRIVERS)/adc/adc_demo/adc_demo.c \
 	$(DRIVERS)/adc/adc_demo/iio_adc_demo.c \
 	$(DRIVERS)/dac/dac_demo/iio_dac_demo.c \
-	$(DRIVERS)/dac/dac_demo/dac_demo.c \
-	$(DRIVERS)/api/irq.c
```
Comment (amiclaus) [inline projects/iio_demo/src.mk:14]: I think irq is required by the aducm3029 platform driver / iio_app

Comment (mchindri) [reply projects/iio_demo/src.mk:14]: and by xilinx uart too

**Comment by APorumb** [comment]
> Can you explain more about the mistake?

@buha warned me about these typos, seemed the only typo was the trailing from line 25

**Review by amiclaus — APPROVED** [review]

**Review by amiclaus — COMMENTED** [review]
By the way, the reference is the commit description is wrong (the line you are changing was added here: https://github.com/analogdevicesinc/no-OS/commit/195fbc0d8f033eeaa0e078a74dd82926f6fe5bd5).

You can add instead a Fixes tag:
Example of fixes tag: https://github.com/analogdevicesinc/linux/commit/ff6c6d527d39bb7e7ea8efa8cd75413f7daa724a

**Comment by APorumb** [comment]
> By the way, the reference is the commit description is wrong (the line you are changing was added here: [195fbc0](https://github.com/analogdevicesinc/no-OS/commit/195fbc0d8f033eeaa0e078a74dd82926f6fe5bd5)).
> 
> You can add instead a Fixes tag: Example of fixes tag: [analogdevicesinc/linux@ff6c6d5](https://github.com/analogdevicesinc/linux/commit/ff6c6d527d39bb7e7ea8efa8cd75413f7daa724a)

Corrected!

**Comment by amiclaus** [comment]
> > By the way, the reference is the commit description is wrong (the line you are changing was added here: [195fbc0](https://github.com/analogdevicesinc/no-OS/commit/195fbc0d8f033eeaa0e078a74dd82926f6fe5bd5)).
> > You can add instead a Fixes tag: Example of fixes tag: [analogdevicesinc/linux@ff6c6d5](https://github.com/analogdevicesinc/linux/commit/ff6c6d527d39bb7e7ea8efa8cd75413f7daa724a)
> 
> Corrected!

Thanks!

**Pull request merged**